internal/controllers/admins.go

```diff
@@ -120,69 +101,50 @@ func (s Server) UpdateUsages(ctx echo.Context) error {
-	usernamefilter := ""
+	var usageData []model.Usage
+	usage := s.GORMDB.Debug().
+		Joins("JOIN user_plans ON user_plans.id = usages.user_plan_id").
+		Joins("JOIN resource_types ON resource_types.id = usages.resource_type_id").
+		Joins("JOIN users ON users.id = user_plans.user_id").
+		Where("cast(now() as date) between user_plans.effective_start_date and user_plans.effective_end_date")
 	if username != "" {
-		usernamefilter = ` and users.username = '` + username + `'`
+		usage.Where("users.user_name = ?", username)
 	}
-	plandata := []AdminUsageDetails{}
-	if err = s.GORMDB.Debug().Raw(
-		`
-			SELECT users.user_name,
-				users.id as user_id,
-				plans.name as plan_name,
-				usages.usage,
-				resource_types.unit,
-				resource_types.name as resource_name
-			FROM user_plans
-			JOIN plans ON plans.id = user_plans.plan_id
-			JOIN usages ON user_plans.id=usages.user_plan_id
-			JOIN quotas ON user_plans.id=quotas.user_plan_id
-			JOIN resource_types ON resource_types.id=quotas.resource_type_id
-			JOIN users ON users.id = user_plans.user_id
-			WHERE cast(now() as date) between user_plans.effective_start_date and user_plans.effective_end_date
-		` + usernamefilter + resourcefilter,
-	).Scan(&plandata).Error; err != nil {
+	if err = usage.Find(&usageData).Error; err != nil {
 		return model.Error(ctx, err.Error(), http.StatusInternalServerError)
 	}
-
-	return ctx.JSON(http.StatusOK, model.SuccessResponse(plandata, http.StatusOK))
+	return model.Success(ctx, usageData, http.StatusOK)
 }
 
-func (s Server) GetAllUserActivePlans(ctx echo.Context) error {
-	username := ctx.Param("username")
-	if username == "" {
-		return ctx.JSON(http.StatusBadRequest, model.ErrorResponse("Invalid UserName", http.StatusBadRequest))
-	}
-	plandata := []PlanDetails{}
-	err := s.GORMDB.Raw(`select plans.name,usages.usage,quotas.quota,resource_types.unit from
-	user_plans
-	join plans on plans.id=user_plans.plan_id
-	join usages on user_plans.id=usages.user_plan_id
-	join quotas on user_plans.id=usages.user_plan_id
-	join resource_types on resource_types.id=quotas.resource_type_id
-	join users on users.id=user_plans.user_id
-	where
-	user_plans.effective_start_date<=cast(now() as date) and
-	user_plans.effective_end_date>=cast(now() as date) ands
-	users.username=?`, username).Scan(&plandata).Error
+func (s Server) GetAllActiveUserPlans(ctx echo.Context) error {
+	var planData []PlanDetails
+	err := s.GORMDB.
+		Table("user_plans").
+		Joins("join plans ON plans.id=user_plans.plan_id").
+		Joins("join usages ON user_plans.id=usages.user_plan_id").
+		Joins("join quota ON user_plans.id=usages.user_plan_id").
+		Joins("join resource_types ON resource_types.id=usages.resource_type_id").
+		Joins("join users ON users.id=user_plans.user_id").
+		Where("user_plans.effective_start_date<=cast(now() as date) " +
+			"AND user_plans.effective_end_date>=cast(now() as date) " +
+			"AND usages.resource_type_id=quota.resource_type_id").
```

Comment:
This is a minor suggestion, but I think that GORM will automatically conjoin multiple where clauses with `AND`, so something like this should be equivalent:

```suggestion
		Where("user_plans.effective_start_date<=cast(now() as date)").
		Where("user_plans.effective_end_date>=cast(now() as date)").
		Where("usages.resource_type_id=quota.resource_type_id").
```